File: src/test/java/no/nav/sosialhjelp/soknad/consumer/sts/FssTokenTest.java
package no.nav.sosialhjelp.soknad.consumer.sts;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class FssTokenTest {

    @Test
    void skalDeserialisereFssToken() throws JsonProcessingException {
        String json = "{\"access_token\":\"asd\",\"token_type\":\"fgh\",\"expires_in\":1234}";

        FssToken fssToken = new ObjectMapper()
                .readerFor(FssToken.class)
                .readValue(json);

        assertThat(fssToken.getAccessToken()).isEqualTo("asd");
        assertThat(fssToken.getTokenType()).isEqualTo("fgh");
        assertThat(fssToken.getExpiresIn().longValue()).isEqualTo(1234L);
    }

}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/client/sts/dto/FssTokenTest.kt
package no.nav.sosialhjelp.soknad.client.sts.dto

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class FssTokenTest {

    @Test
    fun skalDeserialisereFssToken() {
        val json = """{"access_token":"asd","token_type":"fgh","expires_in":1234}"""
        val fssToken = jacksonObjectMapper().readValue<FssToken>(json)
        Assertions.assertThat(fssToken.access_token).isEqualTo("asd")
        Assertions.assertThat(fssToken.token_type).isEqualTo("fgh")
        Assertions.assertThat(fssToken.expires_in).isEqualTo(1234L)
    }
}
